new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java

```diff
@@ -1406,8 +1446,87 @@ public static V1ServiceList listServices(String namespace) {
     }
   }
 
-  // --------------------------- jobs ---------------------------
+  /**
+   * Get V1Service object for the given service name, label and namespace.
+   *
+   * @param serviceName name of the service to look for
+   * @param label  key value pair with which the service is decorated with
+   * @param namespace namespace in which to check for the service
+   * @return V1Service object if found otherwise null
+   * @throws ApiException when there is error in querying the cluster
+   */
+  public static V1Service getService(
+      String serviceName, Map<String, String> label, String namespace)
+      throws ApiException {
+    String labelSelector = null;
+    if (label != null) {
+      String key = label.keySet().iterator().next().toString();
+      String value = label.get(key).toString();
+      labelSelector = String.format("%s in (%s)", key, value);
+      logger.info(labelSelector);
+    }
+    V1ServiceList v1ServiceList
+        = coreV1Api.listServiceForAllNamespaces(
+        Boolean.FALSE, // allowWatchBookmarks requests watch events with type "BOOKMARK".
+        null, // continue to query when there is more results to return.
+        null, // selector to restrict the list of returned objects by their fields
+        labelSelector, // selector to restrict the list of returned objects by their labels.
+        null, // maximum number of responses to return for a list call.
+        Boolean.FALSE.toString(), // pretty print output.
+        null, // shows changes that occur after that particular version of a resource.
+        null, // Timeout for the list/watch call.
+        Boolean.FALSE // Watch for changes to the described resources.
+    );
+    for (V1Service service : v1ServiceList.getItems()) {
+      if (service.getMetadata().getName().equals(serviceName.trim())
+          && service.getMetadata().getNamespace().equals(namespace.trim())) {
+        logger.info("Service Name : " + service.getMetadata().getName());
+        logger.info("Service Namespace : " + service.getMetadata().getNamespace());
+        Map<String, String> labels = service.getMetadata().getLabels();
+        if (labels != null) {
+          for (Map.Entry<String, String> entry : labels.entrySet()) {
+            logger.log(Level.INFO, "Label Key: {0} Label Value: {1}",
+                new Object[]{entry.getKey(), entry.getValue()});
+          }
+        }
+        return service;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Returns NodePort of a admin server service.
+   *
+   * @param serviceName name of admin server service
+   * @param label key value pair with which the service is decorated with
+   * @param namespace namespace in which to check for the service
+   * @return AdminNodePort of the Kubernetes service if exits else -1
+   * @throws ApiException when there is error in querying the cluster
+   */
+  public static int getAdminServiceNodePort(
```

Comment:
Can we not use this method for any server to get the service node port?

Reply:
It can be made generic, but in the case of WebLogic operator, only admin server service has the node port with a name "default".  This method is looking at the node port named as "default"